lib/auth.js

```diff
@@ -270,10 +274,21 @@ internals.Auth.prototype._authenticate = function (request, next) {
         // Check scope
 
         if (config.scope) {
+            var expandScope = function (_, context) {
+
+                return Hoek.reach({
+                    params: request.params,
+                    query: request.query,
+                    payload: request.payload
+                }, context);
+            };
+
+            for (var i = 0, il = config.scope.length; i < il; ++i) {
+                config.scope[i] = config.scope[i].replace(/{([^{}]+)}/g, expandScope);
```

Comment:
You need to set a flag when a route contains a scope with valid `{}` template parameter, then only call this replace logic if that flag is set. Otherwise you are wasting cycles for the 99% not using this feature.